cli/mender-artifact/dump_test.go

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func makeFile(t *testing.T, tmpdir, name, content string) {
+	err := ioutil.WriteFile(path.Join(tmpdir, name), []byte(content), 0644)
+	require.NoError(t, err)
+}
+
+func runAndCollectStdout(args []string) (string, error) {
+	savedStdout := os.Stdout
+	pipeR, pipeW, err := os.Pipe()
+	if err != nil {
+		return "", err
+	}
+
+	os.Stdout = pipeW
+	defer func() {
+		os.Stdout = savedStdout
+		pipeW.Close()
+		pipeR.Close()
+	}()
+
+	var goRoutineErr error
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				goRoutineErr = errors.Errorf("%v", r)
+			}
+			os.Stdout.Close()
+		}()
+
+		goRoutineErr = getCliContext().Run(args)
+	}()
+
+	var printed []byte
+	buf := make([]byte, 1024)
+	for {
```

Comment:
I think this can be replaced with: https://golang.org/pkg/io/ioutil/#ReadAll

Reply:
Right again! I guess I'm writing too much C these days. :-)

Reply:
Just not used to things coming bundled with the language any longer :p